tests/zfs-tests/tests/functional/cli_root/zfs_unshare/zfs_unshare_007_pos.ksh

```diff
@@ -0,0 +1,74 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2016, loli10K. All rights reserved.
+# Copyright (c) 2018 by Delphix. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify that 'zfs destroy' on a shared dataset, will unshare it.
+#
+# STRATEGY:
+# 1. Create and share a dataset with sharenfs.
+# 2. Verify the dataset is shared.
+# 3. Invoke 'zfs destroy' on the dataset.
+# 4. Verify the dataset is not shared.
+#
+
+verify_runnable "global"
+
+function cleanup
+{
+	if datasetexists "$TESTPOOL/$TESTFS/shared1"; then
+		log_must zfs destroy -f $TESTPOOL/$TESTFS/shared1
+	fi
+}
+
+log_assert "Verify 'zfs destroy' will unshare the dataset"
+log_onexit cleanup
+
+# 1. Create and share a datasets with sharenfs.
+log_must zfs create \
+	-o sharenfs=on -o mountpoint=$TESTDIR/1 $TESTPOOL/$TESTFS/shared1
+log_must zfs create \
+	-o sharesmb=on -o mountpoint=$TESTDIR/2 $TESTPOOL/$TESTFS/shared2
```

Comment:
In addition to the other commands, this samba related line will need to go to in order to get the test passing.

```
Test: /usr/share/zfs/zfs-tests/tests/functional/cli_root/zfs_unshare/zfs_unshare_007_pos (run as root) [00:00] [FAIL]
00:27:12.56 ASSERTION: Verify 'zfs destroy' will unshare the dataset
00:27:12.63 SUCCESS: zfs create -o sharenfs=on -o mountpoint=/mnt/testdir/1 testpool/testfs/shared1
00:27:12.70 cannot share 'testpool/testfs/shared2': smb add share failed filesystem successfully created, but not shared
00:27:12.70 ERROR: zfs create -o sharesmb=on -o mountpoint=/mnt/testdir/2 testpool/testfs/shared2 exited 1
00:27:12.70 NOTE: Performing test-fail callback (/usr/share/zfs/zfs-tests/callbacks/zfs_dbgmsg.ksh)
```

Reply:
Sorry, I missed that. It should be fixed now.